src/cdn.py

```diff
@@ -0,0 +1,31 @@
+import requests, requests_cache
+from cache_requests import install_cache_requests
+import logging
+
+LOG = logging.getLogger(__name__)
+install_cache_requests()
+
+def url_exists(url, msid=None):
+    context = {'msid': msid, 'url': url}
+
+    try:
+        resp = requests.head(url)
+    except requests.ConnectionError:
```

Comment:
this seems worthy of an exception? It could mean the CDN is misconfigured. It can be useful to skip it when running the tests of bot-lax-adaptor in isolation, however, since we don't want to depend on external resources.

One way we solved this with IIIF width and height is to return a value that would fail validation, so that the JSON can be generated and partially checked in tests, but it won't go further. Will `None` do?